### ISSUE TITLE
Add deno --deps

### ISSUE THREAD

**Issue by ry** [denoland/deno#630]
Should be one line per dep to start. Don't worry about tree or anything.
```
% deno --deps http://gist.com/blah.js
http://gist.com/blah.js
http://gist.com/dep.js
https://github.com/denoland/deno/master/testing.js
```
@kitsonk please comment on your requirements. The flag parsing can be done in JS, I think? (I wouldn't mind something super hacky to get doing. Like `if (argv.includes("--deps"))`.

We could also parse flags in rust and send it to JS - by following the example of `StartRes.debug_flag`
https://github.com/denoland/deno/blob/ceaf82268282d16b97101c00c75612745de416bb/src/msg.fbs#L82
https://github.com/denoland/deno/blob/ceaf82268282d16b97101c00c75612745de416bb/src/handlers.rs#L141-L149
https://github.com/denoland/deno/blob/ceaf82268282d16b97101c00c75612745de416bb/js/main.ts#L84

Optional: also add `--deps-json` which outputs more info in a structured way - showing the complete tree.

**Comment by kitsonk** [comment]
The compiler currently has this API:

https://github.com/denoland/deno/blob/ceaf82268282d16b97101c00c75612745de416bb/js/compiler.ts#L404-L407

We need to ensure though that when a file is retrieved from deno cache, that the file name listed is the original resolved resource name, and not the cached resource name (which I think it will be).  If that is the case, then likely `os.codeFetch()` API will need to be expanded and the additional information returned from the privileged side and integrated into the `ModuleMetaData` class.

**Comment by ry** [comment]
Fixed in #665
